addOns/commonlib/src/main/java/org/zaproxy/addon/commonlib/timing/TimingUtils.java

```diff
@@ -78,7 +83,8 @@ public static boolean checkTimingDependence(
         int requestsLeft = requestsLimit;
         int requestsMade = 0;
         double secondsLeft = secondsLimit;
-        int currentDelay = 1;
+        int currentDelay = MAX_DELAY - requestsLeft + 1;
```

Comment:
I'm not saying this is wrong but maybe it needs a code comment or something? It caught my eye that delay is altered by number of requests 🤷‍♂️ I'm headed for a meeting and don't have time to dig 😜